server/graphql/loaders/collective.ts

```diff
@@ -113,15 +114,25 @@ export default {
             },
           },
         });
+
+        invitedExpenseUsers = await models.Expense.findAll({
+          attributes: ['FromCollectiveId'],
+          group: ['FromCollectiveId'],
+          raw: true,
+          mapToModel: false,
+          where: { [Op.or]: [{ FromCollectiveId: otherAccountsCollectiveIds }, { status: 'UNVERIFIED' }] },
```

Comment:
:no_entry_sign: The condition here will be translated to the SQL:

```sql
SELECT "FromCollectiveId"
FROM "Expenses"
WHERE "FromCollectiveId" IN (otherAccountsCollectiveIds)
OR status = 'UNVERIFIED'
```

Then we're returning `canSeePrivateInfo` as `invitedExpenseUserIds.has(collectiveId)`, therefore leaking publicly all the emails of users who submitted an expense.

Reply:
The variable name, `invitedExpenseUsers`, is also confusing - this function returns a list of expenses with `FromCollectiveId` set. `invitedExpenseUserIds` contains a list of collective IDs, not user IDs.

Reply:
Sorry, the condition should have an `Op.and` not a `Op.or`. 🤯 

As for the variable names I've renamed them as, `invitedExpenses` and `invitedExpensePayeeIds`.

Reply:
So this new condition translates to:

```sql
SELECT "FromCollectiveId"
FROM "Expenses"
WHERE "FromCollectiveId" IN (otherAccountsCollectiveIds)
AND status = 'UNVERIFIED'
```

Then we're returning `canSeePrivateInfo` as `invitedExpenseUserIds.has(collectiveId)`.

If I translate that to plain English, it gives something like:
> Allow remote user to see private information for all requested accounts who have submitted unverified expenses

We need to be really careful here.

Reply:
Oh I see the problem here; we need a join to return only those accounts which the signed in user has admin privileges. I've added this to the SQL. Let me know if everything looks good.